CopilotKit/examples/next-openai/src/app/headless/page.tsx

```diff
@@ -0,0 +1,390 @@
+          <div className="text-blue-700 font-medium">🔧 Tool 1 Activated</div>
+        </div>
+      );
+    },
+  });
+
+  useCopilotAction({
+    name: "tool2",
+    description: "Tool 2",
+    handler: () => {
+      alert("tool2");
+      return "the secret is 42";
+    },
+  });
+
+  useCopilotAction({
+    name: "tool3",
+    description: "Tool 3",
+    renderAndWaitForResponse: (props) => {
+      return (
+        <div className="bg-gradient-to-r from-purple-50 to-pink-50 border border-purple-200 rounded-xl p-4 my-2">
+          <div className="text-purple-700 font-medium mb-3">🔧 Tool 3 Waiting for Response</div>
+          <button 
+            onClick={() => props.respond?.("the secret is 120")}
+            className="bg-gradient-to-r from-purple-500 to-pink-500 hover:from-purple-600 hover:to-pink-600 text-white px-4 py-2 rounded-lg transition-all duration-200 shadow-md hover:shadow-lg"
+          >
+            Respond
+          </button>
+          <details className="mt-3">
+            <summary className="text-sm text-purple-600 cursor-pointer hover:text-purple-700">Show Details</summary>
+            <div className="mt-2 text-xs bg-purple-100 rounded p-2">{JSON.stringify(props)}</div>
+          </details>
+        </div>
+      );
+    },
+  });
+  
+  useCopilotChatSuggestions({
+    instructions: "Suggest helpful travel-related topics and questions about trip planning, destinations, and travel tips.",
+    minSuggestions: 3,
+  });
+
+  const handleSendMessage = useCallback(() => {
+    chat.appendMessage({
+      id: randomId(),
+      role: "user",
+      content: newMessage,
+    });
+    setNewMessage("");
+  }, [chat, newMessage]);
+
+  const handleSuggestionClick = useCallback((suggestion: { title: string; message: string }) => {
+    chat.appendMessage({
+      id: randomId(),
+      role: "user",
+      content: suggestion.message,
+    });
+  }, [chat]);
+
+  const handleShowDetails = useCallback((message: any) => {
+    setSelectedMessage(message);
+    setIsModalOpen(true);
+  }, []);
+
+  // Manual suggestion examples
+  const setCustomSuggestions = useCallback(() => {
+    chat.setSuggestions([
+      { title: "Plan a trip", message: "Help me plan a 7-day trip to Japan" },
+      { title: "Find hotels", message: "What are the best hotels in Tokyo?" },
+      { title: "Local cuisine", message: "Tell me about Japanese food I must try" }
+    ]);
+  }, [chat]);
+
+  const setCodingSuggestions = useCallback(() => {
+    chat.setSuggestions([
+      { title: "Debug code", message: "Help me debug this React component" },
+      { title: "Code review", message: "Can you review my TypeScript code?" },
+      { title: "Best practices", message: "What are React best practices?" }
+    ]);
+  }, [chat]);
+
+  const clearSuggestions = useCallback(() => {
+    chat.setSuggestions([]);
+  }, [chat]);
+
+  const reloadAISuggestions = useCallback(async () => {
+    await chat.reloadSuggestions();
```

Comment:
When this is called - make sure that we have good error handling if empty adapter